engine/src/main/java/org/terasology/physics/shapes/CollisionShapeFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.physics.shapes;
+
+import org.terasology.math.geom.Vector3f;
+
+import java.util.List;
+
+/**
+ * Factory interface to create new collision shapes.
+ */
+public interface CollisionShapeFactory {
+    /**
+     * Creates a new box shape with the given extents.
+     * @param extents The full extents of the box shape.
+     * @return The box shape.
```

Comment:
This comment is valid for the other methods in this interface: it can't return -the- box shape if it returns a new one every time. I'd rewrite as "A box shape".

Reply:
I understand. How about "The created box shape" for retaining context?